girder/utility/abstract_assetstore_adapter.py

```diff
@@ -144,6 +145,24 @@ def downloadFile(self, file, offset=0, headers=True, endByte=None):
         raise Exception('Must override downloadFile in %s.'
                         % self.__class__.__name__)  # pragma: no cover
 
+    def findInvalidFiles(self, progress=progress.noProgress, filters=None,
+                         checkSize=True, **kwargs):
+        """
+        Finds and yields any invalid files in the assetstore. It is left to
+        the caller to decide what to do with them.
+
+        :param progress: Pass a progress context to record progress.
+        :type progress: :py:class:`girder.utility.progress.ProgressContext`
+        :param filters: Additional query dictionary to restrict the search for
+            files. There is no need to set the ``assetstoreId`` in the filters,
+            since that is done automatically.
+        :type filters: dict or None
+        :param checkSize: Whether to make sure the size of the underlying
+            data matches the size of the file.
+        :type checkSize: bool
+        """
+        return ()
```

Comment:
How about ``raise NotImplementedError()`` here?

I realize that to be consistent, we'd have to change a few other methods too.